### DISCUSSION TITLE
[Solved] "npm run build" failed due to a typescript version check

### DISCUSSION THREAD

**Discussion by LaurentOngaro** [AppOutlet/AppOutlet#229] [Q&A] [answered]
I've followed the steps to build the app but the command "npm run build" failed and raised the following error:
`Error: The Angular Compiler requires TypeScript >=4.0.0 and <4.2.0 but 4.3.5 was found instead.`

The version of typescript I use wad installed with my system: Pop!Os 21.04 (based on ubuntu hirsute 21.04)

**Comment by MessiasLima** [comment]
hello, @LaurentOngaro could you please give more information about your environment? NPM version? How did you install the project dependencies?

**Comment by LaurentOngaro** [comment]
I've followed the instructions you wrote in the readme and in the [page you have recommanded](https://github.com/app-outlet/app-outlet/blob/main/CONTRIBUTING.md).

I've installed the dependencies by running `npm install`

My npm version is 7.5.2

As a try, I've added the line        
`disableTypeScriptVersionCheck": true`
in the tsconfig.json file

This error disappeared, but the build process finally fails with another compilation error and `npm start` fails too

these are the lines at the end of the log file:
```log
22 verbose stack Error: command failed
22 verbose stack     at ChildProcess.<anonymous> (/usr/share/nodejs/@npmcli/promise-spawn/index.js:64:27)
22 verbose stack     at ChildProcess.emit (events.js:314:20)
22 verbose stack     at maybeClose (internal/child_process.js:1022:16)
22 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
23 verbose pkgid app-outlet@2.0.2
24 verbose cwd /mnt/R/Apps_Sources/app-outlet
25 verbose Linux 5.11.0-7620-generic
26 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "build"
27 verbose node v12.21.0
28 verbose npm  v7.5.2
29 error code 127
30 error path /mnt/R/Apps_Sources/app-outlet
31 error command failed
32 error command sh -c npm run build:angular && electron-builder build --linux --publish never
33 verbose exit 127
```

**Comment by MessiasLima** [comment]
its kinda weird because the version of typescript that should be used is defined on the package.json. And it's compatible. Maybe you are using, somehow an external typescript installation.
![Screen Shot 2021-08-20 at 16 48 36](https://user-images.githubusercontent.com/10220064/130259565-2a809e90-ca1d-4552-9ca6-0268a06f6590.png)

If its true, you have to use the version specified. Its not a requirement from the app-outlet application. its a requirement from the angular version that we are using.

**Comment by LaurentOngaro** [comment]
I'm not a typescript user, so I can't understand where the problem is. 
I've just followed the given instructions, no more, no less.
I'll try to restart from scratch, or use a .deb file instead

**Comment by LaurentOngaro** [comment, marked as answer]
I think I've found my error. 
At my first try, I've executed `npm audit fix` before `npm run build` and it broke the dependencies
Without that, the build process is running well.

So my advice... is not to always follow the security advices...

Thanks for your help and your quick answer
